gidgethub/abc.py

```diff
@@ -32,12 +34,15 @@ def __init__(
         oauth_token: Opt[str] = None,
         cache: Opt[CACHE_TYPE] = None,
         base_url: str = sansio.DOMAIN,
+        rate_limiter=None,
     ) -> None:
         self.requester = requester
         self.oauth_token = oauth_token
         self._cache = cache
         self.rate_limit: Opt[sansio.RateLimit] = None
         self.base_url = base_url
+        self.requests_in_flight: int = 0
```

Comment:
Couldn't the rate limiter take care of this? If this was implementing via a subclass then this tracking could be wrapped around `_request()` by the rate limiter and the ABC doesn't have to participate at all.

Reply:
Yeah, that makes sense. will make that change.

Reply:
On a bit more consideration, I think this has to be taken care of by the base class because we want to decrement the inflight count after the request completes.

I think a flaw with the current rate-limit count is that it is the value is reset on every request from github and manually decremented in `_request`.  If you have N in flight, your available limit in `total-N`, but as soon as the first request comes back the rate_limit is reset to `total -1`.  I think it is also possible for it to look like your rate limit is going up if the requests get back to you in a different order than they were processed by github.  Keeping an internal count of the number of un-fulfilled requests "out" and the latest news from github about how many they think we have left is required to have a robust estimate of how many requests we actually have left.

Reply:
Good point, I just hate complicating the base class and expanding its API more.

Is `requests_in_flight` the best name? What about `active_requests`?

And do we need to have a lock in case someone uses this in threads? I'm going to say no since this entire thing is not designed for threading, but the thought did occur to me.

Reply:
It may also be possible to still only the current rate limit as public, but be a bit more careful about managing it.  It would also have to be updated when GH comes back and says the cache is still up-to-date.

Due to the possibility of requests getting re-ordered on the way back from github, I think that to do this reliably we need to make sure that the available count does not go up unless the refresh time also updates and keep an (private?) accounting of the requests that are active.

`active_requests` is a better name.